proposed/http-client/http-client.md

```diff
@@ -46,6 +46,11 @@ interface HttpClient
      *
      * If a request is sent without any prior configuration, a HTTP client MUST NOT follow redirects.
      *
+     * The client MAY do motifications to the Request before sending it. Because PSR-7 objects are
+     * immutable, one cannot assume that the object passed to HttpClient::sendRequest will be the same
+     * object that is actually sent. For example the Request object that is returned by an exception MAY
+     * be a different object than the one passed to sendRequest, so === comparison is not possible.
```

Comment:
lets use reference. depending on how beginner-friendly we want to make it, we could have the === in parenthesis. but go with what you prefer @Nyholm. the important thing is that we mention, and people that are unsure what reference is probably won't get what we try to tell them anyways and have to figure it out the hard way. and anyways all existing PSR-7 implementations will likely already have that behaviour so we just better document the only possible way things can be with PSR-7